test/Swaps.t.sol

```diff
@@ -5,289 +5,327 @@ import "forge-std/Test.sol";
 
 import { PSM } from "../src/PSM.sol";
 
-import { PSMTestBase } from "test/PSMTestBase.sol";
+import { MockERC20, PSMTestBase } from "test/PSMTestBase.sol";
 
-contract PSMSwapAssetZeroToOneTests is PSMTestBase {
+contract PSMSwapFailureTests is PSMTestBase {
 
-    address public buyer    = makeAddr("buyer");
+    address public swapper  = makeAddr("swapper");
     address public receiver = makeAddr("receiver");
 
     function setUp() public override {
         super.setUp();
 
+        // Needed for boundary success conditions
         usdc.mint(address(psm), 100e6);
         sDai.mint(address(psm), 100e18);
     }
 
-    function test_swapAssetZeroToOne_amountZero() public {
+    function test_swap_amountZero() public {
         vm.expectRevert("PSM/invalid-amountIn");
-        psm.swapAssetZeroToOne(0, 0, receiver);
+        psm.swap(address(usdc), address(sDai), 0, 0, receiver);
     }
 
-    function test_swapAssetZeroToOne_receiverZero() public {
+    function test_swap_receiverZero() public {
         vm.expectRevert("PSM/invalid-receiver");
-        psm.swapAssetZeroToOne(100e6, 80e18, address(0));
+        psm.swap(address(usdc), address(sDai), 100e6, 80e18, address(0));
     }
 
-    function test_swapAssetZeroToOne_minAmountOutBoundary() public {
-        usdc.mint(buyer, 100e6);
+    function test_swap_invalid_assetIn() public {
+        vm.expectRevert("PSM/invalid-asset");
+        psm.swap(makeAddr("other-token"), address(sDai), 100e6, 80e18, receiver);
+    }
+
+    function test_swap_invalid_assetOut() public {
+        vm.expectRevert("PSM/invalid-asset");
+        psm.swap(address(usdc), makeAddr("other-token"), 100e6, 80e18, receiver);
+    }
+
```

Comment:
Should add a test to fail when the in and out assets are the same.

Reply:
Added for all three